errorDisplay.js

```diff
@@ -8,60 +8,99 @@ define(function (require, exports, module) {
         ExtensionUtils = brackets.getModule("utils/ExtensionUtils"),
         lineWidgetHTML = require("text!inlineWidget.html"),
         currentErrorWidget;
-
+    var errorToggle;
+    var isShowingDescription;
+    var errorObject;
+    
     ExtensionUtils.loadStyleSheet(module, "main.less");
     require("tooltipsy.source");
 
+    //Publicly available function to remove all errors from brackets
+    function cleanUp(line) {
+        removeButton();
+        removeHighlight(line);
+        hideDescription();
+        isShowingDescription = false;
+    }
+
+    //Publicly available function used to display error markings
+    function scafoldHinter(errorStart, errorEnd, errorObj) {
+        //Setup neccessary variables
+        errorToggle = document.createElement("div");
+        isShowingDescription = false;
+        errorObject = errorObj;
```

Comment:
Instead of having a global variable for this, why not just pass `errorObj` to `showButton(errorObj)` and `highlight(errorObj)`?